File: src/domain/python/validation/validators.py
"""
Validation Layer — Python Edition
"""

def validate_progress_event(evt):
    if not evt.user_id:
        raise ValueError("user_id missing")
    if not isinstance(evt.delta, (float, int)):
        raise ValueError("delta must be numeric")
    return True

def validate_health_score(score):
    if not isinstance(score.value, (float, int)):
        raise ValueError("score.value must be numeric")
    return True

File: src/domain/js/validation/validators.js
/**
 * Validation Layer — JS Edition
 */

export function validateProgressEvent(evt) {
  if (!evt.userId) throw new Error("userId missing");
  if (typeof evt.delta !== "number") throw new Error("delta must be number");
  return true;
}

export function validateHealthScore(score) {
  if (typeof score.value !== "number") throw new Error("value must be number");
  return true;
}
